crypto/rsa/rsa_pk1.c

```diff
@@ -253,3 +256,120 @@ int RSA_padding_check_PKCS1_type_2(unsigned char *to, int tlen,
 
     return constant_time_select_int(good, mlen, -1);
 }
+
+int rsa_padding_check_PKCS1_type_2_TLS(unsigned char *to, size_t tlen,
+                                       const unsigned char *from, size_t flen,
+                                       int client_version, int alt_version)
+{
+    unsigned int i;
+    unsigned int good, found_zero_byte;
+    int zero_index = 0, msg_index, mlen = -1;
+    unsigned char rand_premaster_secret[SSL_MAX_MASTER_KEY_LENGTH];
+    unsigned int version_good;
+    int j;
+
+    /*
+     * If these checks fail then either the message in publicly invalid, or
+     * we've been called incorrectly. We can fail immediately.
+     */
+    if (flen < RSA_PKCS1_PADDING_SIZE + SSL_MAX_MASTER_KEY_LENGTH
+            || tlen < SSL_MAX_MASTER_KEY_LENGTH) {
```

Comment:
I think we wan tlen == SSL_MAX_MASTER_KEY_LENGTH, so fail when tlen != SSL_MAX_MASTER_KEY_LENGTH

Reply:
I did have that in one local version of the code and then changed my mind. Does it actually matter if the to buffer is *bigger* than actually needed? Not sure why we would fail in that case. It also occurred to me that if you *have* to pass SSL_MAX_MASTER_KEY_LENGTH as tlen - do we actually need tlen at all? Anyway I've left this as is for now.

Reply:
The simplification of the code that still used tlen relied on it. Note that you don't use tlen anymore in the code, you've replaced it with SSL_MAX_MASTER_KEY_LENGTH. So with the current code it doesn't matter, it can be bigger.